consensus/core/src/synchronizer.rs

```diff
@@ -266,6 +284,12 @@ impl<C: NetworkClient, V: BlockVerifier, D: CoreThreadDispatcher> Synchronizer<C
 
         let commands_sender_clone = commands_sender.clone();
 
+        if context.parameters.is_sync_last_proposed_block_enabled() {
```

Comment:
That sounds good. Regarding the use of the static once I would suggest to avoid it as we have seen in the past issues that those assumptions can cause to simtests . I believe we can pass down that info from the consensus manager (ex a counter of how many times the consensus has been started in this binary) . Let me address to a follow up PR.